### ISSUE TITLE
Error autocompleting if document changes in-flight

### ISSUE THREAD

**Issue by radeksimko** [radeksimko/terraform-ls#8]
Per LSP spec:

> if a server detects a state change that invalidates the result of a request in execution the server can error these requests with `ContentModified`. If clients receive a ContentModified error, it generally should not show it in the UI for the end-user. Clients can resend the request if appropriate.

We might be able to leverage document versions and compare versions and the beginning and before the end of completion handler.

This is especially important when completing because the operation may be expensive and time-consuming. Getting `github` schema from Terraform can take around `1sec` and could take much more for bigger providers, so the user (or anything outside of the editor) may continue modifying the document in the meantime which then invalidates the list of completion candidates.

**Comment by danieldreier** [comment]
@radeksimko am I understanding right that this is effectively blocked on https://github.com/hashicorp/terraform-ls/issues/426 because with concurrency = 1 `ContentModified` would only be processed after the autocomplete is returned?

**Comment by radeksimko** [comment]
@danieldreier That is right, but fix is coming 🔜  - https://github.com/hashicorp/terraform-ls/pull/489
